vault/resource_generic_secret.go

```diff
@@ -147,6 +146,10 @@ func genericSecretResourceRead(d *schema.ResourceData, meta interface{}) error {
 		if err != nil {
 			return fmt.Errorf("error reading from Vault: %s", err)
 		}
+		if secret == nil {
+			d.SetId("")
```

Comment:
Any time something is removed from state because it wasn't found in a `READ` method, we should log a warning:

```go
if secret == nil {
  log.Printf("[WARN] secret (%s) not found, removing from state", d.Id())
  d.SetId("")
  return nil
}
```